src/ts/component/form/input.tsx

```diff
@@ -226,7 +226,7 @@ class Input extends React.Component<Props, State> {
 			return;
 		};
 
-		this.state.value = String(v || '');
```

Comment:
`||` treats `0` as falsy, similar to `null` which leads to just `''` being stored here.